nix/setup-nix

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+
+if nix --version &> /dev/null; then
+    echo "Nix already installed. Please follow the appropriate steps from the script manually."
+    exit 1
+fi
+
+if apt --version &> /dev/null; then
+    set -x
+    sudo apt-get update -y
+    sudo apt-get install -y nix
+elif pacman --version &> /dev/null; then
+    set -x
+    sudo pacman -S nix
+else
+    echo "Failed to detect package manager."
```

Comment:
```suggestion
    echo "Error: Failed to detect package manager. Please manually install nix."
```
or whatever -- I'd want to make it a bit more clear that this is a hard error, nothing was done, and users will have to install nix manually.